Document support of applicationsets in any namespace (#72955)
openshift/openshift-docs: declarative_clusterconfig/managing-app-sets-in-non-control-plane-namespaces.adoc

:_mod-docs-content-type: ASSEMBLY   
include::_attributes/common-attributes.adoc[]
[id="managing-app-sets-in-non-control-plane-namespaces"]

> = Managing the application set resources in non-control plane namespaces

Reviewer: I think we should combine them?
cc: @svghadi 

```suggestion
= Managing the applicationset resources in non-control plane namespaces
```

Author: @varshab1210 and @svghadi As per this guideline - https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting, we have to keep it separate and we are retaining it as is here.

:context: managing-app-sets-in-non-control-plane-namespaces

toc::[]

:FeatureName: `applicationSet` resources in non-control plane namespaces
include::snippets/technology-preview.adoc[]

> By using application sets, you can automate and manage the deployments of multiple Argo CD applications declaratively from a single mono-repository to many clusters at once with greater flexibility.

Reviewer 2: ```suggestion
By using applicationsets, you can automate and manage the deployments of multiple Argo CD applications declaratively from a single mono-repository to many clusters at once with greater flexibility.
```

Author: @varshab1210 and @svghadi As per this guideline - https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#api-object-formatting, we have to keep it separate and we are retaining it as is here.

> With {gitops-title} 1.12 and later, you can manage the `applicationSet` resources in non-control plane namespaces by explicitly enabling and configuring the Argo CD and `applicationSet` custom resources (CRs) as per your requirements. This functionality is particularly useful in multitenancy environments when you want to manage deployments of Argo CD applications for your isolated teams.

Reviewer: Are back ticks required for applicationset CR references? Can we refer it the same way we refer ArgoCD CR?

cc: @svghadi

Reviewer 2: I have left a suggestion to modify the sentence slightly. The ArgoCD & ApplicationSet are CRs so to highlight them I think we should use `` .

> With {gitops-title} 1.12 and later, you can manage the `applicationSet` resources in non-control plane namespaces by explicitly enabling and configuring the Argo CD and `applicationSet` custom resources (CRs) as per your requirements. This functionality is particularly useful in multitenancy environments when you want to manage deployments of Argo CD applications for your isolated teams.

Author: enabling and configuring the `ArgoCD` and `ApplicationSet` custom resources (CRs)

[NOTE]
====

> The generated Argo CD applications can create resources in any non-control plane namespace. However, the application itself will be in the same namespace as the `applicationSet` resources.

Author: However, the application itself will be in the same namespace as the application set resources.

====

[id="prerequisites_enabling-app-sets-in-any-non-control-plane-namespaces"]
== Prerequisites

* You have a xref:../declarative_clusterconfig/configuring-an-openshift-cluster-by-deploying-an-application-with-cluster-configurations.adoc#using-argo-cd-instance-to-manage-cluster-scoped-resources_configuring-an-openshift-cluster-by-deploying-an-application-with-cluster-configurations[cluster-scoped Argo CD instance].

> * You have link:https://argocd-operator.readthedocs.io/en/latest/usage/apps-in-any-namespace[explicitly enabled and configured] the target namespaces in the Argo CD CR to manage application resources in non-control plane namespaces.

Reviewer 2: ```suggestion
* You have link:https://argocd-operator.readthedocs.io/en/latest/usage/apps-in-any-namespace[explicitly enabled and configured] the target namespaces in the ArgoCD CR to manage application resources in non-control plane namespaces.
```

Author: `ArgoCD` CR

'ApplicationSet` CR vs 1applicationSet` in the spec

> * You have link:https://argocd-operator.readthedocs.io/en/latest/usage/apps-in-any-namespace[explicitly enabled and configured] the target namespaces in the Argo CD CR to manage application resources in non-control plane namespaces.

Reviewer: Do we usually point to the upstream documentation?

Author: @varshab1210 Once the docs for Apps in any namespace are merged, we will replace the link and move this link to the Additional resources section. For now, I am retaining the content as is.

// Enabling the ApplicationSet resources in non-control plane namespaces
include::modules/gitops-enabling-the-application-set-resources-in-non-control-plane-namespaces.adoc[leveloffset=+1]

// Allowing Source Code Manager (SCM) Providers
include::modules/gitops-allowing-scm-providers.adoc[leveloffset=+1]

[role="_additional-resources"]
[id="additional-resources_enabling-app-sets-in-any-non-control-plane-namespaces"]
== Additional resources

> * link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/#the-applicationset-resource[The `applicationSet` resource]
> * link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/Appset-Any-Namespace/[Application set in any namespace]

Reviewer 2: ```suggestion
* link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/#the-applicationset-resource[The `ApplicationSet` resource]
* link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/Appset-Any-Namespace/[ApplicationSet in any namespace]
```

> * link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/Appset-Any-Namespace/[Application set in any namespace]

Reviewer: ```suggestion
* link:https://argo-cd.readthedocs.io/en/release-2.10/operator-manual/applicationset/Appset-Any-Namespace/[ApplicationSet in any namespace]
```

Author: Retaining it as is
